client/js/campaigns/timeline.js

```diff
@@ -33,29 +32,29 @@ function cumulative(data) {
   , total_cents:     0
   }
 
-  var dateDictionary = R.reduce((a,b) => {
-     a[b] = R.merge(proto, {date: b})
+  var dateDictionary = dateStrings.reduce((a,b) => {
+     a[b] = {...proto, date: b }
      return a
-  }, {}, dateStrings)
+  }, {})
 
-  R.reduce((a, b) => {
+  data.reduce((a, b) => {
     a[b.date] = b
     return a
-  }, dateDictionary, data)
+  }, dateDictionary)
   
-  return R.tail(R.reduce((a, b) => {
-    var last = R.last(a)
+  return Object.values(dateDictionary).reduce((a, b) => {
+    var last = a.slice(-1)
```

Comment:
Like before, is there any behavior different for corner cases when the array is empty?

Reply:
Yep, I think similarly here we should use `a[a.length-1]` (I wish there was cleaner JS for this)